### PR TITLE
DOC: reorganize and simplify contributing.rst

### PR DESCRIPTION
**Pull request by tacaswell**
- only provide one installation direction
 - put details about local testing at the top

attn @NelleV 

attn @goldstarwebs

### CODE REVIEW

File: doc/devel/contributing.rst [hunk 1]
```diff
@@ -21,44 +21,62 @@ also welcome to post feature requests or pull requests.
...
+    pip install -v -e ./
 
-.. note::
-  
-    If you decide to do install with ``python setup.py develop`` or ``pip
-    install -v -e``, you will have to rerun::
-   
-      python setup.py build
+This installs Matplotlib in 'editable/develop mode', i.e.,
+builds everything and places symbolic links back to the source code
```
Comment (NelleV) [inline doc/devel/contributing.rst]: I don't think the symbolic link statement is true anymore. It adds a file in which the path to the library is written, and the path is explored at runtime.
(as I don't understand anything about the way python packaging is done, you should take this claim with a grain of salt…)

Comment (tacaswell) [inline doc/devel/contributing.rst]: I think you are right....

File: doc/devel/contributing.rst [hunk 2]
```diff
@@ -21,44 +21,63 @@ also welcome to post feature requests or pull requests.
...
 
+When working on the Matplotlib source, setting up a `virtual
+environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a
+`conda environment <http://conda.pydata.org/docs/using/envs.html>`_ is
+recommended.
+
+.. warning::
+
+   If you already have a version of matplotlib installed, you will need to
```
Comment (QuLogic) [inline doc/devel/contributing.rst]: Matplotlib

File: doc/devel/contributing.rst [hunk 3]
```diff
@@ -21,44 +21,63 @@ also welcome to post feature requests or pull requests.
...
 We use `Git <https://git-scm.com/>`_ for version control and
 `GitHub <https://github.com/>`_ for hosting our main repository.
 
-You can check out the latest sources with the command::
+You can check out the latest sources with the command (see
+:ref:`set-up-fork` for more details)::
 
     git clone git@github.com:matplotlib/matplotlib.git
 
+and navigate to the matplotlib directory.
```
Comment (QuLogic) [inline doc/devel/contributing.rst]: Matplotlib

**Comment by QuLogic** [comment]
I only meant in the newly inserted lines, but that works too. 😛

**Comment by tacaswell** [comment]
It was easier to do a human-moderated matplotlib -> Matplotlib find and replace on the whole file than to sort out which lines I had previously touched.

**Comment by NelleV** [comment]
We need to make sure we are as consistent as possible on that capitalization issue. I'll do that on the whole documentation.

File: doc/devel/contributing.rst [hunk 4]
```diff
@@ -21,44 +21,63 @@ also welcome to post feature requests or pull requests.
...
+When working on the Matplotlib source, setting up a `virtual
+environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a
+`conda environment <http://conda.pydata.org/docs/using/envs.html>`_ is
+recommended.
+
+.. warning::
+
+   If you already have a version of Matplotlib installed, you will need to
+   uninstall it.
```
Comment (anntzer) [inline doc/devel/contributing.rst]: Only in the same venv, I think?  AFAICT that looks big and scary while it is simply not true if the instructions above are followed :-)

Comment (tacaswell) [inline doc/devel/contributing.rst]: You can drive your self to very strange situations with overlapping installs, but will re-word this to be a bit less scary.

File: doc/devel/contributing.rst [hunk 5]
```diff
@@ -21,44 +21,63 @@ also welcome to post feature requests or pull requests.
...
 We use `Git <https://git-scm.com/>`_ for version control and
 `GitHub <https://github.com/>`_ for hosting our main repository.
 
-You can check out the latest sources with the command::
+You can check out the latest sources with the command (see
+:ref:`set-up-fork` for more details)::
 
     git clone git@github.com:matplotlib/matplotlib.git
 
+and navigate to the Matplotlib directory.
```
Comment (dopplershift) [inline doc/devel/contributing.rst]: Matplotlib -> matplotlib since you're talking about the directory.

File: doc/devel/contributing.rst [hunk 6]
```diff
@@ -21,44 +21,65 @@ also welcome to post feature requests or pull requests.
...
+When working on the Matplotlib source, setting up a `virtual
+environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a
+`conda environment <http://conda.pydata.org/docs/using/envs.html>`_ is
+recommended.
+
+.. warning::
+
+   If you already have a version of Matplotlib installed, use an
+   virtual environment or uninstalling using the same method you used
```
Comment (QuLogic) [inline doc/devel/contributing.rst]: uninstalling -> uninstall

**Pull request merged**